### PR TITLE
fix: Update mergify bot config

### PR DESCRIPTION
**Pull request by ValarDragon**
This should make the backport/v8 label work.

I think separately we should switch the repo to either use v8 as our main, or make a new branch called osmo-main, and backport everything to the respective versions.

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: .mergify.yml [hunk 1]
```diff
@@ -2,26 +2,18 @@ pull_request_rules:
   - name: automerge to master with label automerge and branch protection passing
     conditions:
       - "#approved-reviews-by>1"
-      - base=master
+      - base=v0.45.0x-osmo-v7
```
Comment (alexanderbez) [inline .mergify.yml:10]: Can we just have `osmos-main` instead? What happens when we bump past v8? Seems error prone

Comment (ValarDragon) [reply .mergify.yml:10]: Agreed, we should make an `osmo-main`

Comment (alexanderbez) [reply .mergify.yml:10]: How would you like to proceed? Do you want to merge this first and then create the branch or create the the branch later and update the config again?

Comment (ValarDragon) [reply .mergify.yml:10]: Lets merge this first, then create the branch and update it again

**Review by alexanderbez — APPROVED** [review]

**Pull request merged**